node/heartbeat/hearbeatMessageInfo.go

```diff
@@ -5,8 +5,12 @@ import "time"
 // heartbeatMessageInfo retain the message info received from another node (identified by a public key)
 type heartbeatMessageInfo struct {
 	maxDurationPeerUnresponsive time.Duration
-	peerHeartbeats              map[string]*PeerHeartbeat
 	getTimeHandler              func() time.Time
+	timeStamp                   time.Time
+	maxInactiveTime             Duration
+	isActive                    bool
+	shardID                     uint32
+	alreadyAccessed             bool
```

Comment:
Why do we have alreadyAccessed?
Should not update once per execution the data inside the struct.

Reply:
+1

Reply:
removed